### PR TITLE
IT-4439 | Migrate deprecated slot syntax

### PR DESCRIPTION
**Pull request by rogatty**
It's required before we can try Vue 3 migration build: [v3-migration.vuejs.org/migration-build.html#preparations](https://v3-migration.vuejs.org/migration-build.html#preparations)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,97 +1,129 @@
...
-    }],
-    'vue/multi-word-component-names': 0
-  }
-};
\ No newline at end of file
+	plugins: ['import', '@typescript-eslint'],
+	extends: [
+		'eslint:recommended',
+		'plugin:vue/recommended',
+		'prettier',
```
Comment (rogatty) [inline .eslintrc.js:6]: https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md#version-800-2021-02-21

File: .eslintrc.js [hunk 2]
```diff
@@ -1,97 +1,129 @@
...
+		'vue/no-potential-component-option-typo': 'error',
+		'vue/no-static-inline-styles': 'error',
+		'vue/no-template-target-blank': [
+			'error',
+			{
+				allowReferrer: true,
+			},
+		],
+		'vue/v-for-delimiter-style': 'error',
+		'vue/no-undef-components': 'error',
```
Comment (rogatty) [inline .eslintrc.js:126]: https://eslint.vuejs.org/rules/no-unregistered-components.html
> 🚫 This rule was removed in eslint-plugin-vue v9.0.0 and replaced by [vue/no-undef-components](https://eslint.vuejs.org/rules/no-undef-components.html) rule.

File: lib/js/components/Buttons/Button/Button.vue [hunk 3]
```diff
@@ -70,6 +70,7 @@ import {
 import { ICON_BUTTON_COLORS } from '../IconButton/IconButton.consts';
 
 export default {
+	// eslint-disable-next-line vue/no-reserved-component-names
 	name: 'Button',
```
Comment (rogatty) [inline lib/js/components/Buttons/Button/Button.vue:74]: We'll fix it by adding `Ds` prefix which we have plans for.

File: lib/js/components/SurveyQuestions/SurveyQuestionScale/SurveyQuestionScale.spec.ts [hunk 4]
```diff
@@ -29,10 +28,6 @@ describe('SurveyQuestionScale', () => {
 			slots: {
 				explanation,
 			},
-			stubs: {
-				SurveyToggle,
-				Card,
```
Comment (rogatty) [inline lib/js/components/SurveyQuestions/SurveyQuestionScale/SurveyQuestionScale.spec.ts:34]: Alternatively we could use string stubs, something like `<div><slot/><slot name="footer"/></div>`. Using `mount` was easier.

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**